### PR TITLE
On Darwin, pass LIBLTO_PATH to 'ar' and 'lipo' commands from CMake

### PR DESCRIPTION
**Pull request by kubamracek**

### CODE REVIEW

**Comment by kubamracek** [comment]
preset=stdlib_S_standalone_minimal_macho_x86_64,build,test
@swift-ci please test with toolchain and preset

**Comment by kubamracek** [comment]
preset=stdlib_S_standalone_minimal_macho_x86_64,build,test
@swift-ci please test with toolchain and preset

**Comment by kubamracek** [comment]
Alright, this has actually successfully resurrected the 'freestanding'/'minimal' bot (it's green now): https://ci.swift.org/job/swift-PR-stdlib-with-toolchain-osx-preset/113/

File: cmake/modules/UnixCompileRules.cmake [hunk 1]
```diff
@@ -11,3 +11,20 @@ set(CMAKE_C_ARCHIVE_FINISH "")
...
+# When archiving LTO-based .o files with ar/ranlib/libtool on Darwin, the tools
+# use libLTO.dylib to inspect the bitcode files. However, by default the
+# "host" libLTO.dylib is loaded, which might be too old and not understand our
+# just-built bitcode format. So let's instead ask ar/ranlib/libtool to use the
+# just-built libLTO.dylib from the toolchain that we're using to build.
+if(APPLE AND SWIFT_NATIVE_CLANG_TOOLS_PATH)
+  set(liblto_path "${SWIFT_NATIVE_CLANG_TOOLS_PATH}/../lib/libLTO.dylib")
+
+  set(CMAKE_C_ARCHIVE_CREATE "env LIBLTO_PATH=${liblto_path} <CMAKE_AR> crs <TARGET> <LINK_FLAGS> <OBJECTS>")
+  set(CMAKE_C_ARCHIVE_APPEND "env LIBLTO_PATH=${liblto_path} <CMAKE_AR> qs <TARGET> <LINK_FLAGS> <OBJECTS>")
```
Comment (compnerd) [inline cmake/modules/UnixCompileRules.cmake:24]: Could you use `${CMAKE_COMMAND} -E env ...` instead please and the same through out.

**Comment by kubamracek** [comment]
@swift-ci please test

File: cmake/modules/UnixCompileRules.cmake (hunk 1 above)
Comment (kubamracek) [reply cmake/modules/UnixCompileRules.cmake]: Yes, of course. Other than that, looks reasonable?

**Comment by kubamracek** [comment]
preset=stdlib_S_standalone_minimal_macho_x86_64,build,test
@swift-ci please test with toolchain and preset

**Comment by kubamracek** [comment]
Merging to unblock the bot. I'll address any remaining feedback/comments post-merge.

**Pull request merged**